common/qeidentity.c

```diff
@@ -0,0 +1,190 @@
+    0x77, 0xc6, 0x8a, 0x82, 0x9a, 0x00, 0x56, 0xac, 0x8d, 0xed, 0x70,
+    0x14, 0x0b, 0x08, 0x1b, 0x09, 0x44, 0x90, 0xc5, 0x7b, 0xff};
+
+// The isvprodid value of Intel's Production quoting enclave.
+static const uint32_t g_qe_isvprodid = 1;
+
+// The isvsvn value of Intel's Production quoting enclave.
+static const uint32_t g_qeisvsvn = 1;
+
+void dump_info(char* title, uint8_t* data, uint8_t count)
+{
+    OE_TRACE_INFO("%s\n", title);
+    for (uint8_t i = 0; i < count; i++)
+    {
+        OE_TRACE_INFO("[%d] = %x\n", i, data[i]);
+    }
+}
+
+oe_result_t oe_enforce_qe_identity(sgx_report_body_t* qe_report_body)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_get_qe_identity_info_args_t qe_id_args = {0};
+    const uint8_t* pem_pck_certificate = NULL;
+    size_t pem_pck_certificate_size = 0;
+    oe_cert_chain_t pck_cert_chain = {0};
+    oe_parsed_qe_identity_info_t parsed_info = {0};
+
+    OE_TRACE_INFO("Calling %s\n", __PRETTY_FUNCTION__);
+
+    // fetch qe identity information
+    result = oe_get_qe_identity_info(&qe_id_args);
+    if (result == OE_QUOTE_PROVIDER_CALL_ERROR)
+    {
+        // No qe_identity info returned from the quote provider, this could be
+        // because either get_qe_identity_info API was not supported or
+        // unexpected error. In both cases, check against hardcoded fixed quote
+        // enclave properties instead
+        // Assert that the qe report's MRSIGNER matches Intel's quoting
+        // enclave's mrsigner.
+        if (!oe_constant_time_mem_equal(
+                qe_report_body->mrsigner, g_qe_mrsigner, sizeof(g_qe_mrsigner)))
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        if (qe_report_body->isvprodid != g_qe_isvprodid)
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        if (qe_report_body->isvsvn < g_qeisvsvn)
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        // Ensure that the QE is not a debug supporting enclave.
+        if (qe_report_body->attributes.flags & SGX_FLAGS_DEBUG)
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        result = OE_OK;
+        goto done;
+    }
+    OE_CHECK(result);
+
+    // Use QE Identity info to validate QE
+    // Check against fetched qe identityinfo
+    OE_TRACE_INFO("qe_identity.issuer_chain:[%s]\n", qe_id_args.issuer_chain);
+    pem_pck_certificate = qe_id_args.issuer_chain;
+    pem_pck_certificate_size = qe_id_args.issuer_chain_size;
+
+    // validate the cert chain.
+    OE_CHECK(
+        oe_cert_chain_read_pem(
+            &pck_cert_chain, pem_pck_certificate, pem_pck_certificate_size));
+
+    // parse identity info json blob
+    OE_TRACE_INFO("*qe_identity.qe_id_info:[%s]\n", qe_id_args.qe_id_info);
+    OE_CHECK(
+        oe_parse_qe_identity_info_json(
+            qe_id_args.qe_id_info, qe_id_args.qe_id_info_size, &parsed_info));
+
+    // verify qe identity signature
+    OE_TRACE_INFO("Calling oe_verify_tcb_signature\n");
+    OE_CHECK(
+        oe_verify_tcb_signature(
```

Comment:
This does not read well. We are not verifying tcb signature here.

Reply:
Renamed oe_verify_tcb_signature  to oe_verify_ecdsa256_signature to make it neutral